pymc3/model.py

```diff
@@ -920,6 +921,50 @@ def makefn(self, outs, mode=None, *args, **kwargs):
                                    accept_inplace=True,
                                    mode=mode, *args, **kwargs)
 
+    def set_data(self, new_data):
```

Comment:
Does this need to be a method? Can't it just be a function and operate on the passed in `model`? I really want to try to keep the API consistent here, and there is no `model.sample()` or other user-facing methods.